conan/internal/api/export.py

```diff
@@ -19,6 +19,8 @@ def cmd_export(app, hook_manager, global_conf, conanfile_path, name, version, us
                        conanfile.py
     """
     loader, cache = app.loader, app.cache
+    if channel and not user:
```

Comment:
What if ``user`` is specified in the recipe, and ``channel`` in the command line?
This might be breaking.

Reply:
Good catch, will make sure that the error would have been caught in a test too